iceoryx_posh/test/mocks/trigger_handle_mock.hpp

```diff
@@ -47,20 +47,34 @@ class MockTriggeHandle
         return *this;
     }
 
+#ifdef __clang__
```

Comment:
The more I see this `#pragma` sequence the more I think we should define some abbreviation macro for this (or let the compiler ignore it on a global level if possible. Can be decided in a separate PR though, fine for now.